src/cls/async-hooks.ts

```diff
@@ -55,17 +55,18 @@ export class AsyncHooksCLS<Context extends {}> implements CLS<Context> {
     // Create the hook.
     this.hook = this.ah.createHook({
       init: (id: number, type: string, triggerId: number, resource: {}) => {
+        const currentId = this.ah.executionAsyncId();
         // init is called when a new AsyncResource is created. We want code
         // that runs within the scope of this new AsyncResource to see the same
         // context as its "parent" AsyncResource. The criteria for the parent
         // depends on the type of the AsyncResource.
-        if (type === 'PROMISE') {
+        if (type === 'PROMISE' && this.contexts[currentId] !== undefined) {
           // Opt not to use the trigger ID for Promises, as this causes context
           // confusion in applications using async/await.
```

Comment:
Here's my problem with this comment. 'Opt not to use the trigger ID for promise...' comment doesn't explain why the extra `&&` is there in the test.  Nor does the preceding comment explain why the test is the way it is.

I want a comment explaining the extra condition you added.